cmd/cryptdecode/cryptdecode.go

```diff
@@ -43,7 +43,8 @@ See the documentation on the ` + "`crypt`" + ` overlay for more info.
 	Run: func(command *cobra.Command, args []string) {
 		cmd.CheckArgs(2, 11, command, args)
 		cmd.Run(false, false, command, func() error {
-			fsInfo, _, _, config, err := fs.ConfigFs(args[0])
+			fsrc := cmd.NewFsSrc(args)
```

Comment:
`cmd.NewFsSrc(args)` is a relatively expensive operation as it makes the entire backend, so it might involve logging in over the net, which previously we didn't have to do. This will be very noticeable if people are using this to translate 1000 of log entries.

Perhaps what we should do is in the `fs.Name != "crypt"` check, do this bit of extra code to double check that the backend didn't resolve to a crypt backend, so something like

```go
if fsInfo.Name != "crypt" {
    // check for alias and similar backends
    fsrc := cmd.NewFsSrc(args)
    fsInfo, _, _, config, err = fs.ConfigFs(fs.ConfigString(fsrc))
    if err != nil {
        return err
    }
    if fsInfo.Name != "crypt" {
        return errors.New("The remote needs to be of type \"crypt\"")
    }
}
```

I don't know - it is a bit ugly - what do you think?

Reply:
That makes sense and stays consistent with how it's done in [lsjson](https://github.com/rclone/rclone/blob/9e9ead2ac42773ede4d6f1a7282ff3fd0994284b/fs/operations/lsjson.go#L120)
I'll rebase to master and update this commit. Thanks for the feedback!